### PR TITLE
feat(web): add warning when setting a quota superior to the disk size

### PR DESCRIPTION
**Pull request by martabal**
fix for #6691

## Screenshots

Screenshot from 2024-01-30 14-35-29

### CODE REVIEW

**Comment by etnoy** [comment]
Just an idea, but can it display the warning already in the modal, when the number in the text field is too large? 

I.e. like "quota is too large..." next to the text field, shown before you click the button? Hope this makes sense

**Comment by martabal** [comment]
> Just an idea, but can it display the warning already in the modal, when the number in the text field is too large?
> 
> I.e. like "quota is too large..." next to the text field, shown before you click the button? Hope this makes sense

That's even better ! thanks

File: web/src/lib/components/forms/create-user-form.svelte [hunk 1]
```diff
@@ -13,9 +14,17 @@
...
   let isCreatingUser = false;
 
+  $: warning = quotaSize && convertToBytes(Number(quotaSize), 'GiB') > $serverInfo.diskSizeRaw;
+
+  onMount(async () => {
+    if (!$serverInfo) {
+      const { data } = await api.serverInfoApi.getServerInfo();
+      $serverInfo = data;
+    }
+  });
```
Comment (jrasm91) [inline web/src/lib/components/forms/create-user-form.svelte:27]: why is this needed? it seems like it should be initialized in the layout or when the user logs in, not in every component that uses it.

Comment (martabal) [reply web/src/lib/components/forms/create-user-form.svelte]: It feels a bit overkill to have this API request in a layout. I moved it to the page.ts, can you check my last commit ?

Comment (jrasm91) [reply web/src/lib/components/forms/create-user-form.svelte]: Why is it overkill to have it in the layout? That's essentially an entrypoint and means it will only be called once per _page load_.

**Review by jrasm91 — APPROVED** [review]
LGTM

File: web/src/lib/components/forms/create-user-form.svelte (hunk 1 above)
Comment (jrasm91) [reply web/src/lib/components/forms/create-user-form.svelte]: It may be more complicated since it requires auth, so we can look into this again in the future.
